### ISSUE TITLE
detected AUs number is only 8

### ISSUE THREAD

**Issue by Akmend** [CMU-MultiComp-Lab/OpenFace-3.0#9]
The number of detected AUs is only 8, with no correlation. My question is: can the model only detect 8 AUs? Are there other ways to obtain more AU detection results?

**Comment by aish2306** [comment]
I have the same question

**Comment by the0ninjas** [comment]
Same question.

**Comment by johnwilsoniv** [comment]
I have the same question as well. 10 fewer AUs compared to 2.2 and many of the ones they dropped were producing nice outcomes in my research.

**Comment by VictoirZheng** [comment]
Same question here.

**Comment by johnwilsoniv** [comment]
Is anyone else getting non-sensical AU Outputs/complete loss if signal? This is currently unusable for me.

**Comment by brmarkus** [comment]
> Is anyone else getting non-sensical AU Outputs/complete loss if signal? This is currently unusable for me.

What do you mean, can you provide examples, please? Maybe you found more than the eightth AU ;-) ?

**Comment by johnwilsoniv** [comment]
<img width="2384" height="1476" alt="Image" src="https://github.com/user-attachments/assets/fed6450b-c1bf-44d7-a6b8-f9e392aff1f0" />

This figure shows Openface 2.2 vs Openface 3.0 for AU12 on the same video. Openface 2.2 has a clear smile signal but nothing usable for OF3.0

**Comment by ulubilgeulusoy** [comment]
Same question about only having 8 AUs detected without any existing mapping. I do not mind having only 8 AUs, but providing an official mapping similar to OpenFace 2.0 (attached image) would be appreciated so that I can properly cite and use 3.0 in my study.

<img width="1182" height="1243" alt="Image" src="https://github.com/user-attachments/assets/5e87d088-90fe-4c78-80cb-ada51168cf22" />

**Comment by johnwilsoniv** [comment]
Correct me if I am misunderstanding you, but the mapping for [FACS](https://en.wikipedia.org/wiki/Facial_Action_Coding_System) is universal. Not sure if anyone else has been able to successfully use Openface 3.0's AU generation but I am not and the authors didn't respond to my email. I am finishing up a python port of Openface 2.2 with its full AU generation, [stay tuned](https://github.com/johnwilsoniv/pyfaceau).

**Comment by ulubilgeulusoy** [comment]
It is universal, but the OpenFace 3.0 model only outputs 8 AUs. Although I can kinda predict what those 8 are based on my testing, it would be nice for the README file to indicate what those 8 are being generated as output, so it can be properly cited. For example, I generated a very basic custom Python GUI for real-time assessment (see the attached picture, disregarding the numbering of AUs). However, when I try to import all AUs, only 8 of them are displayed, similar to what people mentioned in the initial comments. I'm not sure if I'm missing anything here in terms of outputting AUs, but it would be nice to get a clarification from the authors. 

<img width="399" height="622" alt="Image" src="https://github.com/user-attachments/assets/e1c9e9ae-6121-4339-87a1-95beea558df6" />

**Comment by johnwilsoniv** [comment]
This is the mapping I was using

OPENFACE3_TO_FACS = {
      'AU01_r': 'AU01',  # Inner Brow Raiser
      'AU02_r': 'AU06',  # Cheek Raiser
      'AU03_r': 'AU12',  # Lip Corner Puller (Smile)
      'AU04_r': 'AU15',  # Lip Corner Depressor
      'AU05_r': 'AU17',  # Chin Raiser
      'AU06_r': 'AU02',  # Outer Brow Raiser
      'AU07_r': 'AU09',  # Nose Wrinkler
      'AU08_r': 'AU10',  # Upper Lip Raiser

Based on this line in the training script that pulls from the DISFA dataset:
eval_au = [0, 4, 8, 10, 11, 1, 6, 7]

I agree it's extremely confusing and the authors need to give some clarification on this. The way they label it in the CSV is very confusing.

**Comment by VictoirZheng** [comment]
> This is the mapping I was using
> 
> OPENFACE3_TO_FACS = { 'AU01_r': 'AU01', # Inner Brow Raiser 'AU02_r': 'AU06', # Cheek Raiser 'AU03_r': 'AU12', # Lip Corner Puller (Smile) 'AU04_r': 'AU15', # Lip Corner Depressor 'AU05_r': 'AU17', # Chin Raiser 'AU06_r': 'AU02', # Outer Brow Raiser 'AU07_r': 'AU09', # Nose Wrinkler 'AU08_r': 'AU10', # Upper Lip Raiser
> 
> Based on this line in the training script that pulls from the DISFA dataset: eval_au = [0, 4, 8, 10, 11, 1, 6, 7]
> 
> I agree it's extremely confusing and the authors need to give some clarification on this. The way they label it in the CSV is very confusing.

I'm not sure whether I misunderstood, but I wonder if this map means I can manually "translate" the AU index of the raw 8 element vector generated from this OpenFace-3 into the correct AU index in Ekman's FACS system? I also wonder how these 8 AU data streams extracted from OpenFace-3 (after translation) correlate with the same AU data streams from same videos? 
Besides, considering the confusing results of AUs, what about the emotion recognition results. I wonder is the classifier pretrained based on merely 8 AUs, or on a larger number of AUs?

P.S. Really looking forward to your python port of Openface 2.2 with its full AU generation!

**Comment by johnwilsoniv** [comment]
> > This is the mapping I was using
> > OPENFACE3_TO_FACS = { 'AU01_r': 'AU01', # Inner Brow Raiser 'AU02_r': 'AU06', # Cheek Raiser 'AU03_r': 'AU12', # Lip Corner Puller (Smile) 'AU04_r': 'AU15', # Lip Corner Depressor 'AU05_r': 'AU17', # Chin Raiser 'AU06_r': 'AU02', # Outer Brow Raiser 'AU07_r': 'AU09', # Nose Wrinkler 'AU08_r': 'AU10', # Upper Lip Raiser
> > Based on this line in the training script that pulls from the DISFA dataset: eval_au = [0, 4, 8, 10, 11, 1, 6, 7]
> > I agree it's extremely confusing and the authors need to give some clarification on this. The way they label it in the CSV is very confusing.
> 
> I'm not sure whether I misunderstood, but I wonder if this map means I can manually "translate" the AU index of the raw 8 element vector generated from this OpenFace-3 into the correct AU index in Ekman's FACS system? I also wonder how these 8 AU data streams extracted from OpenFace-3 (after translation) correlate with the same AU data streams from same videos? Besides, considering the confusing results of AUs, what about the emotion recognition results. I wonder is the classifier pretrained based on merely 8 AUs, or on a larger number of AUs?
> 
> P.S. Really looking forward to your python port of Openface 2.2 with its full AU generation!

Anyone who wants to use my python port of Openface 2.2 is welcome to at [this link](https://github.com/johnwilsoniv/pyfaceau) and it's also available on pypi. It has above .95 correlation with openface 2.2 in my testing for all AUs.

**Comment by Joey623** [comment]
It's so confused. I don't know which 8 AUs the predicted 8 AUs correspond to in FACS, and I really hope the authors could clarify this in the README.

**Comment by xxtars** [comment]
I found that the AU labels are defined in the demo2.py script. So the correct labels should be the ones listed here:
https://github.com/CMU-MultiComp-Lab/OpenFace-3.0/blob/main/demo2.py#L189

```
# Example AU descriptions
au_labels = [
    "AU1 - Inner Brow Raiser", "AU2 - Outer Brow Raiser", "AU4 - Brow Lowerer",
    "AU6 - Cheek Raiser", "AU9 - Nose Wrinkler", "AU12 - Lip Corner Puller",
    "AU25 - Lips Part", "AU26 - Jaw Drop"
]
```
